PayPalNativeCheckout/src/main/java/com/braintreepayments/api/PayPalNativeCheckoutClient.java

```diff
@@ -1,14 +1,460 @@
+        this(fragment.getActivity(), fragment.getLifecycle(), braintreeClient, new PayPalNativeCheckoutInternalClient(braintreeClient));
+    }
+
+    /**
+     * Create a new instance of {@link PayPalNativeCheckoutClient} using a {@link BraintreeClient}.
+     * <p>
+     * Deprecated. Use {@link PayPalNativeCheckoutClient (Fragment, BraintreeClient)} or
+     * {@link PayPalNativeCheckoutClient (FragmentActivity, BraintreeClient)}.
+     *
+     * @param braintreeClient a {@link BraintreeClient}
+     */
+    @Deprecated
+    public PayPalNativeCheckoutClient(@NonNull BraintreeClient braintreeClient) {
+        this(null, null, braintreeClient, new PayPalNativeCheckoutInternalClient(braintreeClient));
     }
 
-     public void setListener(PayPalNativeCheckoutListener listener) {
+    @VisibleForTesting
+    PayPalNativeCheckoutClient(FragmentActivity activity, Lifecycle lifecycle, BraintreeClient braintreeClient, PayPalNativeCheckoutInternalClient internalPayPalClient) {
+        this.braintreeClient = braintreeClient;
+        this.internalPayPalClient = internalPayPalClient;
+        if (activity != null && lifecycle != null) {
+            PayPalNativeLifecycleObserver observer = new PayPalNativeLifecycleObserver(this);
+            lifecycle.addObserver(observer);
+        }
+    }
+
+    /**
+     * Add a {@link PayPalNativeCheckoutListener} to your client to receive results or errors from the PayPal flow.
+     * This method must be invoked on a {@link PayPalNativeCheckoutClient (Fragment, BraintreeClient)} or
+     * {@link PayPalNativeCheckoutClient (FragmentActivity, BraintreeClient)} in order to receive results.
+     *
+     * @param listener a {@link PayPalNativeCheckoutListener}
+     */
+    public void setListener(PayPalNativeCheckoutListener listener) {
         this.listener = listener;
-     }
+        if (pendingBrowserSwitchResult != null) {
+            deliverBrowserSwitchResultToListener(pendingBrowserSwitchResult);
+        }
+    }
+
+    private static boolean payPalConfigInvalid(Configuration configuration) {
+        return (configuration == null || !configuration.isPayPalEnabled());
+    }
+
+    private boolean browserSwitchNotPossible(FragmentActivity activity) {
+        return !braintreeClient.canPerformBrowserSwitch(activity, BraintreeRequestCodes.PAYPAL);
+    }
+
+    //TODO NEXT_MAJOR_VERSION
+    //Change link to docs to https://developer.paypal.com/braintree/docs/guides/paypal/overview/android/v4
+    private static Exception createPayPalError() {
+        return new BraintreeException("PayPal is not enabled. " +
+                "See https://developers.braintreepayments.com/guides/paypal/overview/android/ " +
+                "for more information.");
+    }
+
+    //TODO NEXT_MAJOR_VERSION
+    //Change link to docs to https://developer.paypal.com/braintree/docs/guides/client-sdk/setup/android/v4#browser-switch-setup
+    private static Exception createBrowserSwitchError() {
```

Comment:
If these are new exceptions can we change the docs link to the correct link here or do we need to keep the legacy URLs in this new flow?